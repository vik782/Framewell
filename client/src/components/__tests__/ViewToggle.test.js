import React from "react";
import ViewToggle from "../viewToggle";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

// Snapshot test to account for changes to the UI, whether expected or unexpected
test("viewToggle slider renders correctly", () => {
    const viewToggle = renderer.create(<ViewToggle />).toJSON();
    expect(viewToggle).toMatchSnapshot();
});

// Unit Testing
test("viewToggle slider contains initial input value of false", () => {
    const { getByTestId } = render(<ViewToggle />);
    const checkEl = getByTestId("check-viewToggle");
    expect(checkEl.checked).toEqual(false);

});

test("viewToggle slider contains input value of true on click", () => {
    const { getByTestId } = render(<ViewToggle />);
    const checkEl = getByTestId("check-viewToggle");
    const sliderEl = getByTestId("slider-viewToggle");

    fireEvent.click(sliderEl);
    expect(checkEl.checked).toEqual(true);
});