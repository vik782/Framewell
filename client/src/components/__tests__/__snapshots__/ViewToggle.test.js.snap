// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewToggle slider renders correctly 1`] = `
<label
  className="viewToggle"
  data-testid="viewToggle"
>
  <input
    data-testid="check-viewToggle"
    type="checkbox"
  />
  <span
    className="slider"
    data-testid="slider-viewToggle"
  >
    <span />
    <span />
  </span>
</label>
`;
